tools/star.wdl

```diff
@@ -130,67 +530,191 @@ task alignment {
-                    --read_one_fastqs "~{sep(',', read_one_fastqs)}" \
-                    --read_two_fastqs "~{sep(',', read_two_fastqs)}" \
-                    --read_groups "~{read_groups}"
-            else
-                python3 /home/sort_star_input.py \
-                    --read_one_fastqs "~{sep(',', read_one_fastqs)}" \
-                    --read_groups "~{read_groups}"
-            fi
-        else 
-            if [ -n "~{if read_two_fastqs != empty_array then "read_two_fastqs" else ""}" ]
-            then
-                python3 /home/sort_star_input.py \
-                    --read_one_fastqs "~{sep(',', read_one_fastqs)}" \
-                    --read_two_fastqs "~{sep(',', read_two_fastqs)}"
-            else
-                python3 /home/sort_star_input.py \
-                    --read_one_fastqs "~{sep(',', read_one_fastqs)}"
-            fi
-        fi
+        # odd constructions a combination of needing white space properly parsed
+        # and limitations of the WDL v1.0 spec
+        python3 /home/sort_star_input.py \
+            --read-one-fastqs "~{sep(',', read_one_fastqs)}" \
+            ~{if (read_two_fastqs != empty_array) then "--read-two-fastqs" else ""} "~{
+                sep(',', (
+                    if (read_two_fastqs != empty_array)
+                    then read_two_fastqs
+                    else []
+                ))
+            }" \
+            ~{if defined(read_groups) then "--read-groups" else ""} "~{
+                if defined(read_groups)
+                then read_groups
+                else ''
+            }"
 
-        STAR --readFilesIn \
-                $(cat read_one_fastqs_sorted.txt) \
-                $(cat read_two_fastqs_sorted.txt) \
-             --readFilesCommand "gunzip -c" \
-             --genomeDir ~{star_db_dir} \
-             --runThreadN "$n_cores" \
-             --outSAMunmapped Within \
-             --outSAMstrandField intronMotif \
-             --outSAMtype BAM Unsorted \
-             --outSAMattributes NH HI AS nM NM MD XS \
-             --outFilterMultimapScoreRange 1 \
-             --outFilterMultimapNmax 20 \
-             --outFilterMismatchNmax 10 \
-             --alignIntronMax 500000 \
-             --alignMatesGapMax 1000000 \
-             --sjdbScore 2 \
-             --alignSJDBoverhangMin 1 \
-             --outFilterMatchNminOverLread 0.66 \
-             --outFilterScoreMinOverLread 0.66 \
-             --outFileNamePrefix ~{prefix + "."} \
-             --twopassMode Basic \
-             --limitBAMsortRAM ~{memory_limit_bytes} \
-             --outSAMattrRGline $(cat read_groups_sorted.txt)
+        read -ra read_one_args < read_one_fastqs_sorted.txt
+        read -ra read_two_args < read_two_fastqs_sorted.txt
+        read -ra read_group_args < read_groups_sorted.txt
+        STAR --readFilesIn "${read_one_args[@]}" "${read_two_args[@]}" \
+            --readFilesCommand "gunzip -c" \
```

Comment:
Not to add complications, but should we be assuming that `read_one_fastqs` contains gzipped files? We could rename it to `read_one_fastqs_gz` for consistency.

Reply:
Or we could parameterize the read command.

Reply:
This was a good catch. Path of least resistance is go back to using the `_gz` suffix and specifying we expect gzipped files. That would be easier (and less error prone) than dealing with multiple/arbitrary read commands.